stonesoup/resampler/particle.py

```diff
@@ -42,7 +43,23 @@ def resample(self, particles):
         u_j = u_i + (1 / n_particles) * np.arange(n_particles)
         index = weight_order[np.searchsorted(cdf, np.log(u_j))]
         new_particles = Particles(state_vector=particles.state_vector[:, index],
-                                  weight=[weight]*n_particles,
+                                  weight=[weight] * n_particles,
                                   parent=Particles(state_vector=particles.state_vector[:, index],
                                                    weight=particles.weight[index]))
         return new_particles
+
+
+class ESSResampler(SystematicResampler):
```

Comment:
I wonder if this class could be a wrapper that could be used with other resamplers.
```python
class ESSResampler(Resampler):
    resampler: Resampler = Property(doc="Resampler to wrap, which is called when ESS below threshold")
    ...
```